### ISSUE TITLE
Bug in log generation from process tree containing loops

### ISSUE THREAD

**Issue by Macuyiko** [process-intelligence-solutions/pm4py#147]
Let's say we generate a process tree as follows:

```
from pm4py.algo.simulation.tree_generator import factory as tree_gen_factory
from pm4py.visualization.process_tree import factory as pt_vis_factory
from pm4py.objects.conversion.process_tree import factory as pt_conv_factory
from pm4py.visualization.petrinet import factory as pn_vis_factory
from pm4py.objects.process_tree import semantics

parameters = {
    'loop': 1.0, 'mode': 5, 'min': 1, 'max': 5, 'parallel': 0.0, 'choice': 1.0, 'silent': 0.0
}

tree = tree_gen_factory.apply(parameters=parameters)
```

E.g. using the following tree

![image](https://user-images.githubusercontent.com/271897/76429988-14a56900-63b0-11ea-8774-987456e1d436.png)

And the corresponding Petri net:

![image](https://user-images.githubusercontent.com/271897/76430017-1d963a80-63b0-11ea-981a-26a9e41027ec.png)

Now trying to generate traces from this always results in the following:

```
log = semantics.generate_log(tree, no_traces=1)
print([e['concept:name'] for e in log[0]])

# ['a']
```

The issue seems to be that after encountering a loop operator, no parts of the tree remain enabled except for the first child. The generation cuts of after generating the first loop structure.

As an additional remark, it might be a good idea to add a parameter to the log generation in terms of how many times each loop branch can be visited at most, to prevent infinite generation.

**Comment by fit-alessandro-berti** [comment]
Thanks for signaling.

Actually, the problem has been found and fixed.

The next release will solve that

**Comment by fit-alessandro-berti** [comment]
The problem has been resolved in release 1.2.12
